### ISSUE TITLE
Inconsistency in expected labels for linearity measures

### ISSUE THREAD

**Issue by Lechemi** [w4k2/problexity#10]
L1 measure expects classes to be 1 and -1, while L3 expects 1 and 0. Should be standardized and documented.

**Comment by JKomorniczak** [comment]
Can you provide more information? 
The intend of the current implementation was to compute measures for labels [0,1] for all classification measures. This is verified by the tests and seems to be working fine. 
Keep in mind L1 is also implemented for regression tasks, and therefore will work for labels [-1,1].

**Comment by Lechemi** [comment]
I'm referring to classification measures.

Inside l1's definition there's this line:
```python
loss = np.sum( np.max((np.zeros((len(y_mistakes))), 1-(y_mistakes*fun)), axis=0))
``` 

The problem with it is that it assumes that classes are -1 and +1, since it's using the hinge loss.
In a 0/1 scenario, all samples with class 0 that get misclassified by the SVM would have a contribution of exactly 1 in the sum.
I've also empirically verified that l1's output is slightly different between -1/+1 and 0/1 scenarios.

In general, I think expected classes should be explicitly documented, whether they're 0 and 1 or -1 and +1. For instance, calling l3 with classes -1 and +1 will result in a confusing ValueError.

**Comment by JKomorniczak** [comment]
Thank you for providing more insight and for assessing the measures in detail.

We examined the implementation in reference to the article introducing the measure (*Lorena, Ana C., et al. "How complex is your classification problem? A survey on measuring classification complexity." ACM Computing Surveys (CSUR) 52.5 (2019): 1-34.*). It seems like the previous implementation was not accurate. 
The current one uses the `scikit-learn` decision function to calculate the distance to the SVM hyperplane and should reliably express the L1 measure. We do not currently compute the loss function manually, and the class labels of [0,1] are handled by the used classification model (`LinearSVC`) for hinge loss calculation.

We additionally updated the documentation concerning the expected labels — for the `ComplexityCalculator` class, those may represent a multiclass problem, whereas, for individual measures, a binary classification task with labels [0,1]. 

We hope this improves the clarity when using the package. Thank you again for noticing such a latent error.
